firmware/user/modes/mode_ddr.c

```diff
@@ -0,0 +1,876 @@
+ *
+ * @param arg unused
+ */
+static void ICACHE_FLASH_ATTR ddrAnimateNotes(void* arg __attribute__((unused)))
+{
+    ddr->NoteIdx = (ddr->NoteIdx + 1) % 4;
+    // testUpdateDisplay();
+}
+
+static void ICACHE_FLASH_ATTR ddrUpdateDisplay(void* arg __attribute__((unused)))
+{
+    switch(ddr->mode)
+    {
+        default:
+        case DDR_MENU:
+        {
+            drawMenu(ddr->menu);
+            break;
+        }
+        case DDR_SCORE:
+        {
+            clearDisplay();
+            char perfectsText[24];
+            char hitsText[24];
+            char missesText[24];
+
+            ets_snprintf(perfectsText, sizeof(perfectsText),   "Perfects:    %03d", ddr->perfects);
+            ets_snprintf(hitsText, sizeof(perfectsText),       "Okays:       %03d", ddr->okays);
+            ets_snprintf(missesText, sizeof(perfectsText),     "Misses:      %03d", ddr->misses);
+
+            if (ddr->didLose)
+            {
+                plotText(38,5, "You died", IBM_VGA_8, WHITE);
+            }
+            else
+            {
+                plotText(38,5, "You win!", IBM_VGA_8, WHITE);
+            }
+
+            plotText(36,30, perfectsText, TOM_THUMB, WHITE);
+            plotText(36,40, hitsText, TOM_THUMB, WHITE);
+            plotText(36,50, missesText, TOM_THUMB, WHITE);
+            break;
+        }
+        case DDR_GAME:
+        {
+            //Game update logic
+            ddrHandleArrows();
+
+            // Clear the display
+            clearDisplay();
+
+            //ddrUpdateButtons();
+            ddrArrowRow* curRow;
+            ddrArrow* curArrow;
+            int curStart;
+            int curCount;
+            int curEnd;
+
+            for (int rowIdx = 0; rowIdx < 4; rowIdx++)
+            {
+                curRow = &(ddr->arrowRows[rowIdx]);
+                curStart = curRow->start;
+                curCount = curRow->count;
+                curEnd = (curStart + curCount) % ARROW_ROW_MAX_COUNT;
+
+                for(int arrowIdx = curStart; arrowIdx != curEnd; arrowIdx = (arrowIdx + 1) % ARROW_ROW_MAX_COUNT)
+                {
+                    curArrow = &(curRow->arrows[arrowIdx]);
+                    
+                    
+                    drawPngSequence(&ddrSkullSequenceHandle, (curArrow->hPos-400)/12, 48 - rowIdx * 16, false, false, 0, ddr->NoteIdx);
+                }
+            }
+
+            plotCircle(110, 55-16*3, BTN_RAD-3, WHITE); // Top
+            plotCircle(110, 55-16*3-3, 2, WHITE); // subcircle
+
+            plotCircle(110, 55-16*2, BTN_RAD-3, WHITE);
+            plotCircle(110-3, 55-16*2, 2, WHITE); // subcircle
+
+            plotCircle(110, 55-16*1, BTN_RAD-3, WHITE);
+            plotCircle(110+3, 55-16*1, 2, WHITE); // subcircle
+
+            plotCircle(110, 55-16*0, BTN_RAD-3, WHITE); // Bottom
+            plotCircle(110, 55-16*0+3, 2, WHITE); // subcircle
+
+            /*
+            if(ddr->ButtonDownState & UP)
```

Comment:
Aww, why is this commented out? I think the feedback could be useful. Or maybe instead have a little explosion sprite or something when a note is hit? Just having the sprites disappear on a hit doesn't quite feel feedback enough to me